src/lib/cacheDatabase/index.ts

```diff
@@ -0,0 +1,239 @@
+/**************************************************************************
+ *  (C) Copyright ModusBox Inc. 2020 - All rights reserved.               *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       Yevhen Kyriukha - yevhen.kyriukha@modusbox.com                   *
+ *                                                                        *
+ *  CONTRIBUTORS:                                                         *
+ *       James Bush - james.bush@modusbox.com                             *
+ **************************************************************************/
+
+import knex, { Knex } from 'knex';
+import Cache from './cache';
+import SDK from '@mojaloop/sdk-standard-components';
+
+const cachedFulfilledKeys: string[] = [];
+const cachedPendingKeys: string[] = [];
+
+interface UserInfo {
+  displayName?: string;
+  firstName?: string;
+  lastName?: string;
+}
+
+const getName = (userInfo: UserInfo) =>
+  userInfo && (userInfo.displayName || `${userInfo.firstName} ${userInfo.lastName}`);
+
+const getTransferStatus = (data: any) => {
+  if (data.currentState === 'succeeded') {
```

Comment:
`data.currentState` should be compared against enums, although I am not sure if this was the original code or if it's your code?

I.e. if this was the original code, then we can handle this as a follow up technical debt item.

This applies to the rest of the `currentState` comparison below. Also, take note of the inconsistencies of the states being compared.

Some of them are lower-case and other comparisons are using upper-case...not sure if that is an issue?